pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -375,3 +375,13 @@ def _do_check_save(self, filepath, current, epoch, trainer, pl_module):
         for cur_path in del_list:
             if cur_path != filepath:
                 self._del_model(cur_path)
+
+    def on_save_checkpoint(self, trainer, pl_module):
+        return {
+            'best_model_score': self.best_model_score,
+            'best_model_path': self.best_model_path,
```

Comment:
what do you think, is it better to have it like this or use the constants for these strings? When I added these constants recently, my motivation was that the hard coded strings could lead to bugs when changed in one place but not another, because I saw them used inside trainer as well.

Reply:
yeah my hope is now that we have the state persistence logic as part of the callback, we shouldn't need to have this logic reaching across multiple parts of the codebase. i think it should now be self-contained within the callback.